### PR TITLE
txn model- connecting ledgers in transaction snark statement

### PR DESCRIPTION
**Pull request by deepthiskumar**
Added two ledger hashes that are assumed to be connecting the first pass and second pass ledgers in a transaction statement based on the merge rules described here- https://github.com/MinaProtocol/mina/discussions/12000
Checklist:

- [ ] Modified the current draft of release notes with details on what is completed or incomplete within this project
- [ ] Document code purpose, how to use it
  - Mention expected invariants, implicit constraints
- [ ] Tests were added for the new behavior
  - Document test purpose, significance of failures
  - Test names should reflect their purpose
- [ ] All tests pass (CI will check this if you didn't)
- [ ] Serialized types are in stable-versioned modules
- [ ] Does this close issues? List them

* Closes #0000

### CODE REVIEW

File: src/lib/transaction_snark/transaction_snark.ml [hunk 1]
```diff
@@ -136,6 +136,8 @@ module Statement = struct
               , 'pending_coinbase
               , 'local_state )
               Registers.Stable.V1.t
+          ; connecting_ledger_left : 'ledger_hash
```
Comment (nholland94) [inline src/lib/transaction_snark/transaction_snark.ml:139]: These values need to be "nullable". Can these be options? Or is it easier (for inside the snark) to treat a sigil values such as a hash of `0` to be the "null" value?

**Review by nholland94 — APPROVED** [review]
Looks good, but have a question regarding setting null values for the connecting ledger hashes.

File: src/lib/transaction_snark/transaction_snark.ml (hunk 1 above)
Comment (deepthiskumar) [reply src/lib/transaction_snark/transaction_snark.ml:139]: I suppose we could use zero. Can it not be the target ledger hash? For payments, I used target hash of first pass ledger for second pass ledgers and connecting ledgers

**Comment by deepthiskumar** [comment]
!ci-build-me

**Review by nholland94 — APPROVED** [review]

**Comment by deepthiskumar** [comment]
Changes merged into #12548

**Pull request closed without merging**